dev/io.openliberty.netty.internal.tls.impl/src/io/openliberty/netty/internal/tls/impl/NettyTlsProviderImpl.java

```diff
@@ -0,0 +1,327 @@
+    
+    /**
+     * DS deactivate
+     * 
+     * @param ctx
+     * @param reason
+     */
+    @Activate
+    protected void activate(ComponentContext ctx){
+        if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+            Tr.debug(tc, "NettyTlsProviderImpl activate " + this);
+        }
+    }
+    
+    /**
+     * DS deactivate
+     * 
+     * @param ctx
+     * @param reason
+     */
+    @Modified
+    protected void modified(ComponentContext ctx){
+        if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+            Tr.debug(tc, "NettyTlsProviderImpl modified " + this);
+        }
+    }
+
+
+    /**
+     * DS deactivate
+     * 
+     * @param ctx
+     * @param reason
+     */
+    @Deactivate
+    protected void deactivate(ComponentContext ctx, int reason){
+        if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+            Tr.debug(tc, "NettyTlsProviderImpl deactivate " + this);
+        }
+    }
+
+    /**
+     * Build a {@link io.netty.handler.ssl.SslContext} for an outbound connection
+     * 
+     * @param Map<String, Object> sslOptions
+     * @param String host
+     * @param String port
+     * @return SslContext
+     */
+    public SslContext getOutboundSSLContext(Map<String, Object> sslOptions, String host,
+            String port) {
+
+        SSLContext jdkContext;
+        try {
+        	Properties props = createProps(sslOptions);
+            String alias = (String)props.getProperty(ALIAS_KEY);
+            jdkContext = getSSLContext(alias, props, false, host, port, port, null);
+            if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                Tr.debug(tc, "getOutboundSSLContext SSLContext:", jdkContext);
+            }
+        } catch (Exception e) {
+            if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                Tr.debug(tc, "getOutboundSSLContext exception caught creating SSLContext: " + e);
+            }
+            return null;
+        }
+
+        try {
+            SslContext nettyContext = new JdkSslContext(jdkContext, true, null, SupportedCipherSuiteFilter.INSTANCE,
+                    null, ClientAuth.OPTIONAL, null, false);
+            return nettyContext;
+        } catch (Exception e) {
+            if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                Tr.debug(tc, "getOutboundSSLContext exception caught creating JdkSslContext: " + e);
+            }
+            return null;
+        }
+    }
+
+    /**
+     * Build a {@link io.netty.handler.ssl.SslContext} for an intbound connection
+     * 
+     * @param Map<String, Object> sslOptions
+     * @param String host
+     * @param String port
+     * @return SslContext
+     */
+    public SslContext getInboundSSLContext(Map<String, Object> sslOptions, String host, String port) {
+
+        SSLContext jdkContext;
+        try {
+        	// TODO: Check alias being sent here
```

Comment:
Can't check in TODO's ? Raise an issue etc?

Reply:
Issue created to follow up https://github.com/OpenLiberty/open-liberty/issues/23424